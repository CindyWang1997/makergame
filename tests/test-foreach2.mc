extern void end_game();

aide { create { printstr("aide"); } }

main {
  bool created;
  create {
    int i;
    create aide;
    i = 1;
    foreach(aide h)
      if (i < 5) { create aide; i = i + 1; }
    /* should create 5 aides, since new one should also be iterated over */
    end_game();
  }
}

