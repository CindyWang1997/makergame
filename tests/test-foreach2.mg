extern void end_game();

object aide { event create { printstr("aide"); } }

object main {
  bool created;
  event create {
    int i;
    create aide;
    i = 1;
    foreach(aide h)
      if (i < 5) { create aide; i = i + 1; }
    /* should create 5 aides, since new one should also be iterated over */
    end_game();
  }
}

