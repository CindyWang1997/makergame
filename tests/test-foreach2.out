aide
aide
aide
aide
aide
